src/main/java/com/example/restapi/controller/EmployeeController.java

```diff
@@ -0,0 +1,60 @@
+package com.example.restapi.controller;
+
+import com.example.restapi.entities.Employee;
+import com.example.restapi.services.EmployeeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1")
+public class EmployeeController {
+
+	@Autowired
+	private EmployeeService employeeService;
+
+	@GetMapping("/employee")
+	public List<Employee> retrieveAllEmployees() {
+		return this.employeeService.findAll();
+	}
+
+	@GetMapping("/employee/{id}")
+	public Optional<Employee> retrieveEmployee(@PathVariable Long id) {
+		return employeeService.findById(id);
+	}
+
+	@PutMapping("/employee/{id}")
+	public ResponseEntity<Object> updateEmployee(@PathVariable Long id, @RequestBody Employee employee) {
+		Optional<Employee> empFromDb = this.employeeService.findById(id);
+		if (empFromDb.isPresent()) {
+			employee.setId(id);
+			this.employeeService.save(employee);
+			return ResponseEntity.ok().body(employee);
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@DeleteMapping("/employee/{id}")
+	public ResponseEntity deleteEmployee(@PathVariable Long id) {
+		this.employeeService.deleteById(id);
+		return ResponseEntity.ok().build();
+	}
+
+	@PostMapping("/employee")
+	public ResponseEntity createEmployee(@RequestBody Employee employee) {
```

Comment:
Separate API Request Payload from Repo Entity, we won't use our Repo Enitty as a Payload
example: I only allow to update only `firstName`, the entity has all the fields which mean when I do some request, I must fill in all of them, there is no way to tell I am updating only firstName.

Best practice: Create a class called: `EmployeeUpdatePayload` which include only fields we allow to update, then map that `EmployeeUpdatePayload` to `Employee` entity then save.

Tip: Mapping fields will be painfull, to reduce tons lines of code just for getter and setter, consider using ModelMapper Library --> https://www.baeldung.com/entity-to-and-from-dto-for-a-java-spring-application

Reply:
Fixed